status/flowcell.py

```diff
@@ -299,42 +306,50 @@ def fetch_barcodes(self, run_name):
             return None
 
     
-    def fetch_args(self, name):
+    def fetch_args(self, run_name):
+
         view_args = self.application.nanopore_runs_db.view("info/args", descending=True)
-        l = [row for row in view_args.rows if name in row.key][0].value
+        rows = view_args[run_name].rows
 
-        group = "([^\s=]+)"                             # Text component of cmd argument
-        
-        flag_pair = re.compile(f"^--{group}={group}$")  # Double-dash argument with assignment
-        flag_key_or_header = re.compile(f"^--{group}$") # Double-dash argument w/o assignment
-        pair = re.compile(f"^(?!--){group}={group}$")   # Non-double-dash argument with assignment
-        val = re.compile(f"^(?!--){group}$")            # Non-double-dash argument w/o assignment
-
-        entries = {}
-        idxs_args = iter(zip(range(0, len(l)), l))
-        for (idx, arg) in idxs_args:
-            # Flag pair --> Tuple
-            if re.match(flag_pair, arg):
-                k, v = re.match(flag_pair, arg).groups()
-                entries[(k,v)]="pair"
-            # Flag single
-            elif re.match(flag_key_or_header, arg):
-                # If followed by pair --> Header
-                if re.match(pair, l[idx+1]):
-                    h = re.match(flag_key_or_header, arg).groups()[0]
-                    entries[h]="header"
-                # If followed by val --> Add both as tuple and skip ahead
-                elif re.match(val, l[idx+1]):
-                    k = re.match(flag_key_or_header, arg).groups()[0]
-                    v = re.match(val, l[idx+1]).groups()[0]
-                    entries[(k,v)]="pair"
-                    next(idxs_args)
-            # Pair 
-            elif re.match(pair, arg):
-                k, v = re.match(pair, arg).groups()
-                entries[(k,v)]="sub_pair"
+        if rows:
 
-        return entries
+            args = rows[0].value
+
+            group = "([^\s=]+)"                             # Text component of cmd argument
+            
+            flag_pair = re.compile(f"^--{group}={group}$")  # Double-dash argument with assignment
+            flag_key_or_header = re.compile(f"^--{group}$") # Double-dash argument w/o assignment
+            pair = re.compile(f"^(?!--){group}={group}$")   # Non-double-dash argument with assignment
+            val = re.compile(f"^(?!--){group}$")            # Non-double-dash argument w/o assignment
+
+            entries = {}
```

Comment:
You could have this statement outside the `if` and return `entries` outside it, then you don't need an 
`else:
            return None`

Reply:
The same applies for fetch_barcodes(), but I think having the `else: return None` is better in terms of readability compared to returning an empty dict.